packages/design-system/src/components/Badge/Badge.test.tsx

```diff
@@ -1,42 +1,50 @@
 import React from 'react';
-import Badge from './Badge';
+import Badge, { BadgeProps } from './Badge';
 import { render, screen } from '@testing-library/react';
 
-describe('Badge', () => {
-  it('should render a default badge', () => {
-    render(<Badge>Foo</Badge>);
-    expect(screen.getByText('Foo')).toMatchSnapshot();
-  });
-
-  it('should render an info badge', () => {
-    render(<Badge variation="info">Foo</Badge>);
-    expect(screen.getByText('Foo')).toMatchSnapshot();
-  });
+import register from 'preact-custom-element';
+register(Badge, 'ds-badge');
+declare global {
+  namespace JSX {
+    interface IntrinsicElements {
+      'ds-badge': BadgeProps;
+    }
+  }
+}
 
-  it('should render a success badge', () => {
-    render(<Badge variation="success">Foo</Badge>);
-    expect(screen.getByText('Foo')).toMatchSnapshot();
-  });
+const defaultProps = {
+  children: 'Foo',
+};
+function renderBadge(props = {}) {
+  // return render(<Badge {...defaultProps} {...props} />);
+  return render(<ds-badge {...defaultProps} {...props}></ds-badge>);
+}
 
-  it('should render a warning badge', () => {
-    render(<Badge variation="warn">Foo</Badge>);
-    expect(screen.getByText('Foo')).toMatchSnapshot();
+describe('Badge', () => {
+  // inserting multiples of Foo
+  it('should render a default badge', () => {
+    renderBadge();
+    const badge = screen.getByText('Foo');
+    expect(badge).toMatchSnapshot();
   });
 
-  it('should render an alert badge', () => {
-    render(<Badge variation="alert">Foo</Badge>);
-    expect(screen.getByText('Foo')).toMatchSnapshot();
+  it('applies variation classes', () => {
+    renderBadge({ variation: 'success' });
+    const badge = screen.getByText('Foo');
+    expect(badge.classList).toContain('ds-c-badge--success');
   });
 
   it('should render a big badge', () => {
-    render(<Badge size="big">Foo</Badge>);
+    renderBadge({ size: 'big' });
     const badge = screen.getByText('Foo');
     expect(badge.className).toContain('ds-c-badge--big');
   });
 
-  it('should render custom classNames', () => {
-    render(<Badge className="super-special-badge">Foo</Badge>);
+  // adding class-name multiple times?
```

Comment:
same multiples problem. `class-name` and `classname` appear on several nodes within component and i can't figure out why.